ir/src/symbol_table.rs

```diff
@@ -151,6 +159,33 @@ impl SymbolTable {
         Ok(())
     }
 
+    /// Adds all random values by their identifier names and array length.
+    pub(super) fn insert_random_values(
+        &mut self,
+        values: &RandomValues,
+    ) -> Result<(), SemanticError> {
+        self.num_random_values = values.size() as u16;
+        let mut offset = 0;
+
+        let a = Some(10);
+        if a.is_some() {}
+
+        // add array name to the symbol table
+        self.insert_symbol(
+            values.name(),
+            IdentifierType::RandomValuesBinding(offset, values.size() as usize),
+        )?;
+        // add array values to the symbol table
```

Comment:
nit: I think these comments could be slightly clearer. I'd probably do the following:

```
...
        // add the name of the random values array to the symbol table
        self.insert_symbol(
            values.name(),
            IdentifierType::RandomValuesBinding(offset, values.size() as usize),
        )?;
        // add the named random value bindings to the symbol table
...
```